devtools/cli/src/main/java/io/quarkus/cli/build/MavenRunner.java

```diff
@@ -0,0 +1,174 @@
+
+    final OutputOptionMixin output;
+    final Path projectRoot;
+
+    public MavenRunner(OutputOptionMixin output, Path projectRoot) {
+        this.output = output;
+        this.projectRoot = projectRoot;
+        verifyBuildFile();
+    }
+
+    @Override
+    public File getWrapper() {
+        return ExecuteUtil.findWrapper(projectRoot, windowsWrapper, otherWrapper);
+    }
+
+    @Override
+    public File getExecutable() {
+        return ExecuteUtil.findExecutable("mvn",
+                "Unable to find the maven executable, is it in your path?",
+                output);
+    }
+
+    @Override
+    public Path getProjectRoot() {
+        return projectRoot;
+    }
+
+    @Override
+    public OutputOptionMixin getOutput() {
+        return output;
+    }
+
+    @Override
+    public BuildTool getBuildTool() {
+        return BuildTool.MAVEN;
+    }
+
+    QuarkusProject quarkusProject() {
+        return MavenProjectBuildFile.getProject(projectRoot, output, Version::clientVersion);
+    }
+
+    @Override
+    public Integer listExtensions(RunModeOption runMode, ListFormatOptions format, boolean installable, String searchPattern)
+            throws Exception {
+        // we do not have to spawn to list extensions for maven
+        QuarkusCommandOutcome outcome = new ListExtensions(quarkusProject(), output)
+                .fromCli(true)
+                .all(false)
+                .installed(!installable)
+                .format(format.getFormatString())
+                .search(searchPattern)
+                .execute();
+
+        return outcome.isSuccess() ? CommandLine.ExitCode.OK : CommandLine.ExitCode.SOFTWARE;
+    }
+
+    @Override
+    public Integer addExtension(RunModeOption runMode, Set<String> extensions) throws Exception {
+        AddExtensions invoker = new AddExtensions(quarkusProject());
+        invoker.extensions(extensions);
+        return invoker.execute().isSuccess() ? CommandLine.ExitCode.OK : CommandLine.ExitCode.SOFTWARE;
+    }
+
+    @Override
+    public Integer removeExtension(RunModeOption runMode, Set<String> extensions) throws Exception {
+        RemoveExtensions invoker = new RemoveExtensions(quarkusProject());
+        invoker.extensions(extensions);
+        return invoker.execute().isSuccess() ? CommandLine.ExitCode.OK : CommandLine.ExitCode.SOFTWARE;
+    }
+
+    @Override
+    public BuildCommandArgs prepareBuild(BuildOptions buildOptions, RunModeOption runMode, List<String> params) {
+        ArrayDeque<String> args = new ArrayDeque<>();
+        setMavenProperties(args, runMode.isBatchMode());
+
+        if (runMode.isBatchMode()) {
+            args.add("-B");
+        }
+
+        if (buildOptions.clean) {
+            args.add("clean");
+        }
+        args.add("install");
+
+        if (buildOptions.buildNative) {
+            args.add("-Dnative");
+        }
+        if (buildOptions.skipTests()) {
+            setSkipTests(args);
+        }
+        args.add("install");
```

Comment:
This is duplicated

Reply:
I will add this to the list of follow on nit issues... i tested this at some point and it worked, but I realize I didn't add it to the actual test.. (the auto-generated help for maps does show that equals sign.. it should work either way)